lib/spack/spack/test/test_activations.py

```diff
@@ -78,7 +43,23 @@ def create_dir_structure(tmpdir, dir_structure):
 
 
 @pytest.fixture()
-def python_and_extension_dirs(tmpdir):
+def builtin_and_mock_packages():
+    # These tests use mock_repo packages to test functionality of builtin
+    # packages. They therefore put the mock repo UNDER the builtin repo.
+    old_path = spack.repo.path
+    repo_dirs = [spack.paths.packages_path, spack.paths.mock_packages_path]
```

Comment:
I likely should have mentioned it when I first wrote these tests, but IMO it's worth mentioning that this is to test functionality defined in the builtin `Python`/`Perl` packages (which is why for example we don't just use the mock `Python` package).

Reply:
The comment on line 47 addresses this directly

Reply:
I read that; it specifies "builtin" packages but doesn't mention `Python` or `Perl` which are the primary packages we want to test, and I think that detail would be useful. I think it's worth explicitly saying "this uses a mock python extension to test the behavior of the builtin Python package".

Also, I think specifying priority is clearer than saying "under" (i.e. "prioritize the builtin repo over the mock repo")